src/BotPlutusInterface/CardanoCLI.hs

```diff
@@ -142,19 +142,23 @@ calculateMinUtxo ::
   Map DatumHash Datum ->
   TxOut ->
   Eff effs (Either Text Integer)
-calculateMinUtxo pabConf datums txOut =
-  join
-    <$> callCommand @w
-      ShellArgs
-        { cmdName = "cardano-cli"
-        , cmdArgs =
-            mconcat
-              [ ["transaction", "calculate-min-required-utxo", "--alonzo-era"]
-              , txOutOpts pabConf datums [txOut]
-              , ["--protocol-params-file", pabConf.pcProtocolParamsFile]
-              ]
-        , cmdOutParser = mapLeft Text.pack . parseOnly UtxoParser.feeParser . Text.pack
-        }
+calculateMinUtxo pabConf datums txOut = do
+  let outs = txOutOpts pabConf datums [txOut]
+  case outs of
+    [] -> pure $ Left "When constructing the transaction, no output values were specified."
```

Comment:
I think this check should also be in `buildTx`.
Additional thing to consider, is that there will be no `calculateMinUtxo` in Vasil compliant version, so check here will be no more and `build-raw` in `buildTx` will start to fail.

Reply:
Currently, the check for empty list is redundant because, when we call `calculateMinUtxo` from `calculateMinUtxos` (present in `Balance.hs`) there's always some value present in the txOut. But it nevertheless make sense to handle this case, in case anyone tries to call `calculateMinUtxo` where the `txOutValue` is not always non-zero. But after vasil compliance of BPI this function would be redundant, and so would be the need to check for if `txOutValue` is non-zero.

Reply:
@IAmPara0x 
> Currently, the check for empty list is redundant because, when we call calculateMinUtxo from calculateMinUtxos (present in Balance.hs) there's always some value present in the txOut.

Why does it always have some value there? Because of minimum Ada adjustment?

Reply:
We need to call `calculateMinUtxo` when we want to get minimum about of lovelace that should be present in a `TxOut` of a `Tx`, this is required during balancing. But, if there's a `TxOut` with zero value then we don't need to consider it as an actual output to a Tx.

Reply:
I do agree that this check should be moved to `buildTx`.

@IAmPara0x This check was added for the specific case in which a transaction would only contain a zero `Value` as the output. Which would result in an empty list from the changes that I made here:

```haskell
txOutOpts :: PABConfig -> Map DatumHash Datum -> [TxOut] -> [Text]
 txOutOpts pabConf datums =
   concatMap
     ( \TxOut {txOutAddress, txOutValue, txOutDatumHash} ->
         if Value.isZero txOutValue
           then []
           else
```

Reply:
@MitchyCola oh, I get it now. Thanks for pointing it out!.

Reply:
Since the error is originating from the use of `calculateMinUtxos` inside of `balanceTxLoop` (in `Balance.hs`), the error gets thrown before the use of `buildTx`. Therefore, I decided to move the check into `preBalancedTx`.

Reply:
What if just put check in both `calculateMinUtxos` and `buildTx`? It will keep this commit in working state, and vasil update will just remove  `calculateMinUtxos`.

It feels like omitting check that result of `txOutOpts` can return `[]` will shoot one day.

Reply:
Fixed